en/advanced_config/prearm_arm_disarm.md

```diff
@@ -4,13 +4,42 @@ Vehicles may have moving parts, some of which are potentially dangerous when pow
+- **Pre-armed:** Motors/propellers are locked but actuators for non-dangerous electronics are powered (e.g. ailerons, flaps etc.).
+- **Armed:** Vehicle is fully powered. Motors/propellers may be turning (dangerous!)
 
-A [safety switch](../getting_started/px4_basic_concepts.md#safety_switch) (if available) is used as a precondition for arming the vehicle, and may also be used to enable the pre-armed state. When permitted, arming is enabled using an arming sequence, switch or MAVLink command. 
+A [safety switch](../getting_started/px4_basic_concepts.md#safety_switch) (if available) is used as a precondition for arming the vehicle, and may also be used to enable the pre-armed state.
 
-PX4 allows you to configure how prearming, arming and disarming work using parameters (which can be edited in *QGroundControl* via the [parameter editor](../advanced_config/parameters.md)), as described in the following sections.
+When permitted, arming is enabled using an [arming gesture](#arm_disarm_gestures), [arming switch](#arm_disarm_switch) or MAVLink command. 
+
+PX4 allows you to configure how pre-arming, arming and disarming work using parameters (which can be edited in *QGroundControl* via the [parameter editor](../advanced_config/parameters.md)), as described in the following sections.
+
+> **Note** Arming/disarming parameters can be found in [Parameter Reference > Commander](../advanced_config/parameter_reference.md#commander) (search for `COM_ARM_` and `COM_DISARM_*`).
+
+## Arming Gesture {#arm_disarm_gestures}
+
+By default the vehicle is armed and disarmed by moving the RC throttle/yaw stick into a particular position and holding for for 1 second:
+- **Arming:** stick to *bottom right*.
+- **Disarming:** stick to *bottom left*. 
+
+The hold time can be configured using [COM_RC_ARM_HYST](#COM_RC_ARM_HYST).
+
+Parameter | Description
+--- | ---
+<span id="COM_RC_ARM_HYST"></span>[COM_RC_ARM_HYST](../advanced_config/parameter_reference.md#COM_RC_ARM_HYST) | Time RC stick must be held in arm/disarm position before arming/disarming occurs (default: 1 second).
+
+## Arming Switch {#arm_disarm_switch}
+
+An *Arm switch* can be configured to trigger arm/disarm, and will then be used instead of a [gesture](#arm_disarm_gestures) (which is disabled).
+
+The switch is assigned (and enabled) using [RC_MAP_ARM_SW](#RC_MAP_ARM_SW).
+The behaviour of the switch is then further configured using [COM_ARM_SWISBTN](#COM_ARM_SWISBTN).
+
+Parameter | Description
+--- | ---
+<span id="RC_MAP_ARM_SW"></span>[RC_MAP_ARM_SW](../advanced_config/parameter_reference.md#RC_MAP_ARM_SW) | Arm switch channel (default: 0 - unassigned). If defined the specified RC channel is used instead of stick.
+<span id="COM_ARM_SWISBTN"></span>[COM_ARM_SWISBTN](../advanced_config/parameter_reference.md#COM_ARM_SWISBTN) | Arm switch is only a button. 0: Arm switch is a switch that stays on when armed, 1: Arm switch is a button that only triggers arming and disarming.
```

Comment:
@MaEtUgR Don't understand at all what COM_ARM_SWISBTN is supposed to do - it what the difference between switch and button you are trying to imply is.

Is it that if set as a switch (0=default) the button state is locked? - so essentially if you move switch to armed state the arming command is "held down". Presumably this means that
-  if you can't arm you'll get a stream of messages "can't arm" back from the vehicle?
- in this case to disarm you move back to the disarm state.

It could alternatively be that when used as a switch you move to the position and hold for at least 1 second, wait for arming or disarming, then move back to the off position when that completes.

While for value of 1 perhaps it is a transitory switch. So command sent once. If you set to arm and it doesn't work, to try again you have to toggle back to disarm (command ignored) and then try again.

Can you confirm which/what?

Reply:
> the button state is locked?
No, it's handled differently you have to retry again either way just on the switch it's the transition that triggers a command on the button it's holding it for the same amount of time like the stick gesture that triggers a command once.

Summary:
- If the arm switch RC channel **is a switch** set `COM_ARM_SWISBTN` 0 and
the **transition** of the switch position from disarmed to armed commands arm,
the transition from armed to disarm commands disarm.
If the command is rejected the switch is out of sync with the arm state and does not retry.
- If the arm switch RC channel **is a button or momentary switc**h set `COM_ARM_SWISBTN` 1
**holding the button down** while disarmed does the same as holding the stick arm gesture,
holding it down when armed does the same as holding the stick disarm gesture.
If you press the button indefinitely it only does the first thing no retry, no arm, disarm, arm disarm.

I consider the switch implementation is not that useful for a position-controlled flying product but rather for hobby pilots flying more manual modes e.g. FPV racing because in acro the disarm command usually succeeds and it's kind of your kill/disarm failsafe. While the button implementation is meant for hover platforms like bigger drones with a remote that has a button e.g. like the H520/ST16 has.

Does that help? 😇